be/src/vec/exec/scan/scanner_scheduler.cpp

```diff
@@ -232,12 +233,18 @@ void ScannerScheduler::_scanner_scan(ScannerScheduler* scheduler, ScannerContext
         auto block = ctx->get_free_block(&get_free_block);
         status = scanner->get_block(state, block, &eos);
         VLOG_ROW << "VOlapScanNode input rows: " << block->rows() << ", eos: " << eos;
-        if (!status.ok()) {
+        // The VFileScanner for external table may try to open not exist files,
+        // Because FE file cache for external table may out of date.
+        if (!status.ok() && (typeid(*scanner) == typeid(doris::vectorized::VFileScanner) &&
+                             !status.is_not_found())) {
             LOG(WARNING) << "Scan thread read VOlapScanner failed: " << status.to_string();
             // Add block ptr in blocks, prevent mem leak in read failed
             blocks.push_back(block);
             break;
         }
+        if (status.is_not_found()) {
```

Comment:
I think we should move this logic into the `VFileScanner::_get_next_reader()`?
And `VFileScanner::_get_block_impl` will return `eos = true` when file is not found